.github/workflows/ci-actions.yml

```diff
@@ -1,8 +1,29 @@
 name: Jakarta Contexts and Dependency Injection CI
 
 on:
+  push:
+    branches-ignore:
+      - 'wip/**'
+      - 'draft/**'
+    paths-ignore:
+      - '.gitignore'
+      - 'CONTRIBUTING.adoc'
+      - 'NOTICE.md'
+      - 'LICENSE'
+      - '*.md'
+      - '*.adoc'
+      - '*.txt'
   pull_request:
-    branches: [ master ]
+    types: [opened, synchronize, reopened, ready_for_review]
```

Comment:
I don't think we want CI to trigger for any PR - `main` branch would be sufficient.
In case we want to push to older version, that will have to have its own CI setup, mainly due to different JDK versions it is supposed to support.

Reply:
The main purpose of triggering the CI on a push or pull request is to be sure that every thing is building well before making a pull request, means the building matrix is fine and the tests succeeds, this will give time to contributor to fix things before making a pull request and have an idea about the ci execution result even on a custom branch.

For the merge request on main branch,  the CI always needs approval from maintainers